packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -99,16 +99,105 @@ async function promiseRaceWithWinner<T>(promises: Promise<T>[]): Promise<{ index
+
+    // Tracks all blob IDs evicted from cache
+    private readonly blobsEvicted: Set<string> = new Set();
+
+    // Initial time-out to purge data from cache
+    // If this time out is very small, then we purge blobs from cache too soon and that results in a lot of
+    // requests to storage, which brings down perf and may trip protection limits causing 429s
+    // Also we need to ensure that buildCachesForDedup() is called with full cache for summarizer client to build
+    // its SHA cache for blobs (currently that happens as result of requesting snapshot tree)
+    private blobCacheTimeoutDuration = 2 * 60 * 1000;
+
+    public get value() {
+        return this._blobCache;
+    }
+
+    public addBlobs(blobs: IBlob[]) {
+        blobs.forEach((blob) => {
+            assert(blob.encoding === "base64" || blob.encoding === undefined);
+            this._blobCache.set(blob.id, blob);
+        });
+        this.scheduleClearBlobsCache();
+    }
+
+    /**
+     * Schedule a timer for clearing the blob cache or defer the current one.
+     */
+    private scheduleClearBlobsCache() {
+        if (this.blobCacheTimeout !== undefined) {
+            // If we already have an outstanding timer, just signal that we should defer the clear
+            this.deferBlobCacheClear = true;
+        } else {
+            // If we don't have an outstanding timer, set a timer
+            // When the timer runs out, we'll decide whether to proceed with the cache clear or reset the timer
+            const clearCacheOrDefer = () => {
+                this.blobCacheTimeout = undefined;
+                if (this.deferBlobCacheClear) {
+                    this.deferBlobCacheClear = false;
+                    this.scheduleClearBlobsCache();
+                } else {
+                    // NOTE: Slightly better algorithm here would be to purge either only big blobs,
+                    // or sort them by size and purge enough big blobs to leave only 256Kb of small blobs in cache
+                    // Purging is optimizing memory footprint. But count controls potential number of storage requests
+                    // We want to optimize both - memory footprint and number of future requests to storage.
+                    // Note that Container can realize data store or DDS on-demand at any point in time, so we do not
+                    // control when blobs will be used.
+                    //
+                    // That all said, SPO does not keep old snapshots around for long, so we are running chances of not
+                    // being able to rehydrate data store / DDS in the future if we purge anything (and with blob de-duping,
+                    // even if blob read by runtime, it could be read again in the future)
+                    // So for now, purging is disabled.
+                    // this._blobCache.forEach((_, blobId) => this.blobsEvicted.add(blobId));
```

Comment:
Maybe rather than commenting out, put a flag on the `BlobCache` to turn it on or off (default off)?